elastic/detection-rules
# [Hunt Tuning] Enforce `STATS` or `KEEP` functions in ES|QL hunting queries

# Pull Request

*Issue link(s)*:
* https://github.com/elastic/detection-rules/pull/4146



## Summary - What I changed

* Updated all queries that did not have `keep` or `stats` in them to include at least `keep`
* Added `validate_esql_query` method to `Hunt` dataclass that is called only in a query starts with `from` (assuming all ES|QL queries for hunting will use `FROM` source commands only



## How To Test
Run: `pytest tests/test_hunt_data.py` from this branch to check for success. Then make adjustments to any hunt queries that are ES|QL and remove the Keep or Stats functions.

<img width="1873" alt="Screenshot 2024-10-15 at 7 18 19 PM" src="https://github.com/user-attachments/assets/b61337fc-284d-4e5e-8687-61b5ff2f2e50">



## Checklist




## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?
